src/Template.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace App;
+
+use App\Template\Context;
+
+/**
+ * Template engine created for bugs.php.net. Goal is not to reinvent a new
+ * template engine and compete with other much better ones out there but to
+ * provide a vanilla PHP approach and separate application logic and presentation
+ * layers a bit.
+ *
+ * Engine provides several methods to create main template layout, sections, and
+ * escaping of strings to not introduce too common XSS vulnerabilities.
+ */
+class Template
+{
+    /**
+     * Templates directory contains all application templates.
+     * @var string
+     */
+    private $templatesDir;
+
+    /**
+     * Default context of template engine (usage of $this in template belongs to
+     * set context).
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * Pool of registered functions in the application.
+     * @var array
+     */
+    private $functions = [];
+
+    /**
+     * Pool of global variables.
+     * @var array
+     */
+    private $variables = [];
+
+    /**
+     * Class constructor.
+     */
+    public function __construct(string $templatesDir, Context $context)
+    {
+        $this->templatesDir = $templatesDir;
+        $this->context = $context;
+    }
+
+    /**
+     * This enables adding new variables in the template scope right after
+     * initializing a template engine. Some variables in templates are like
+     * parameters or globals and should be added only on one place instead of
+     * repeating them at each ...->render() call.
+     */
+    public function add(array $vars = []): void
```

Comment:
Should be renamed to assign() because you do "assign" a value to a place holder inside of the template, you do not add a variable to code.
```suggestion
    public function assign(array $vars = []): void
```

Reply:
Thanks, applied. :+1: